### PR TITLE
libadwaita 1.4.0 appstream 0.16.3 (new formula)

### PR DESCRIPTION
**Pull request by danyeaw**
This PR builds on and replaces #142989 and the excellent work by @chenrui333 getting the initial formula for appstream created.

- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

We have one error for `brew audit --new appstream` since it includes patches that are accepted upstream, but a new version of appstream hasn't yet been released.

### CODE REVIEW

**Comment by danyeaw** [comment]
@chenrui333 I checked my commit messages a few times. Do I need to squash our commits together to make the CI happy?

Will it be possible to merge this with `brew audit --new appstream` failing for patches existing or are we going to have to wait for a new appstream version to be released?

**Comment by chenrui333** [comment]
> @chenrui333 I checked my commit messages a few times. Do I need to squash our commits together to make the CI happy?

yeah, mostly for commit style, but the audit failures are not quite related :) 

> Will it be possible to merge this with brew audit --new appstream failing for patches existing or are we going to have to wait for a new appstream version to be released?

ideally yes, but since this is blocking the other formulae, i guess we can proceed with accepting the upstream patches

File: Formula/lib/libadwaita.rb [hunk 1]
```diff
@@ -25,6 +25,8 @@ class Libadwaita < Formula
     sha256 x86_64_linux:   "9c9ed668d719bab47cc21c920d3214de8c81b9447232ba8f0bce968053a41b05"
   end
 
+  depends_on "appstream" => [:build, :test]
```
Comment (chenrui333) [inline Formula/lib/libadwaita.rb:28]: this needs to be runtime dep?
```suggestion
  depends_on "appstream"
```

Comment (danyeaw) [reply Formula/lib/libadwaita.rb]: Thanks @chenrui333, great catch!

**Comment by chenrui333** [comment]
looks like needs gperf for linux build

```
  Program gperf found: NO
  
  meson.build:183:8: ERROR: Program 'gperf' not found or not executable
```

**Comment by danyeaw** [comment]
> looks like needs gperf for linux build

Thanks, I pushed a new commit with it as a build dependency on Linux for appstream.

**Comment by chenrui333** [comment]
also need gettext for linux build

**Comment by chenrui333** [comment]
I think we should be good after it

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
The linux build should be good now.

**Review by chenrui333 — APPROVED** [review]

File: Formula/lib/libadwaita.rb [hunk 2]
```diff
@@ -25,12 +25,14 @@ class Libadwaita < Formula
...
 
+  depends_on "cmake" => :build
   depends_on "gettext" => :build
   depends_on "gobject-introspection" => :build
   depends_on "meson" => :build
   depends_on "ninja" => :build
   depends_on "pkg-config" => [:build, :test]
   depends_on "vala" => :build
+  depends_on "appstream"
   depends_on "gtk4"
```
Comment (chenrui333) [inline Formula/lib/libadwaita.rb:37]:
```suggestion

  uses_from_macos "python" => :build
```

Comment (chenrui333) [reply Formula/lib/libadwaita.rb:37]: cc @danyeaw

Comment (danyeaw) [reply Formula/lib/libadwaita.rb:37]: Hey @chenrui333, does `uses_from_macos` work for Linux failures?

Comment (chenrui333) [reply Formula/lib/libadwaita.rb:37]: yes

Comment (chenrui333) [reply Formula/lib/libadwaita.rb:37]: here is PR ref, https://github.com/Homebrew/homebrew-core/pulls?q=is%3Apr+uses_from_macos+%22python%22

Comment (danyeaw) [reply Formula/lib/libadwaita.rb:37]: Thanks @chenrui333, I pushed a hopefully final update with this change. I just got confused with the term uses_from_macos, but I understand it is more about using the system Python.

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
Looks good, merging! @danyeaw Thanks again! 🎉

**Comment by danyeaw** [comment]
Thanks @chenrui333! ❤️

**Pull request merged**